### PR TITLE
Disable clock_gettime workaround for OSX version 10.12.

### PR DESCRIPTION
**Pull request by yfguo**
OSX has clock_gettime() starting from version 10.12. The workaround
breaks the build on OSX 10.12. The patch checks the version of current
OSX and disables the workaround if the version is greater or equal to
10.12.

Signed-off-by: Yanfei Guo yguo@anl.gov

### CODE REVIEW

**Comment by jsquyres** [comment]
Since libfabric already has a configure script, it would likely be better to just test for the presence of `clock_gettime()`.

While sometimes unavoidable, checking versions ends up being a nightmare in the long run, and should be avoided whenever possible.

**Comment by yfguo** [comment]
Good point. I can do it in that way.

**Comment by jsquyres** [comment]
@yfguo Excellent; many thanks.

**Comment by jsquyres** [comment]
@shefty We need this for v1.4, otherwise libfabric is busted on the latest OSX.  If @yfguo isn't able to come up with a PR in the Very Near Future, @bturrubiates might make a new PR with the alternate fix (i.e., minor update to configury).

**Comment by bturrubiates** [comment]
> @shefty We need this for v1.4, otherwise libfabric is busted on the latest OSX. If @yfguo isn't able to come up with a PR in the Very Near Future, @bturrubiates might make a new PR with the alternate fix (i.e., minor update to configury).

I've submitted #2493, @yfguo could you verify?

**Comment by shefty** [comment]
If we can get this in, great.  Sayantan checked the build for < 10.12.  Hopefull Yanfei can test 10.12, as I don't know anyone with that version.

**Comment by bturrubiates** [comment]
> If we can get this in, great. Sayantan checked the build for < 10.12. Hopefull Yanfei can test 10.12, as I don't know anyone with that version.

I can test on my personal laptop, but not until I get home. So, at 6ish.

**Comment by yfguo** [comment]
Hey folks. Sorry for the late reply. I have pushed another fix. Now it is using configure to check the availability of clock_gettime and turn on/off the workaround. I have also tested https://github.com/ofiwg/libfabric/pull/2493 and it works fine on my OSX 10.12.

File: include/osx/osd.h [hunk 1]
```diff
@@ -46,27 +46,41 @@
...
 #ifdef _POSIX_HOST_NAME_MAX
 #define HOST_NAME_MAX _POSIX_HOST_NAME_MAX
 #else
 #define HOST_NAME_MAX 255
 #endif
 
-typedef int clockid_t;
+#define pthread_yield pthread_yield_np
+
+#define bswap_64 OSSwapInt64
```
Comment (jsquyres) [inline include/osx/osd.h]: What are the defines for `pthread_yield` and `bswap_64` for?  They don't look related to this `clock_gettime` fix...?

Comment (shefty) [inline include/osx/osd.h]: It looks like those defines were just relocated in the file to move them outside the get clock define.

Comment (yfguo) [inline include/osx/osd.h]: You are right. I relocated the code to put the `clock_gettime` related codes together

**Comment by jsquyres** [comment]
Fixed and credited in 8de3943.

**Comment by wesbland** [comment]
According to https://github.com/Homebrew/homebrew-core/pull/6551, this looks like it's still broken on El Capitan (MacOS 10.11). Homebrew is still on 1.3 because of it.
